product/views.py

```diff
@@ -73,5 +72,95 @@ def get(self, request):
+            
+            color_list = list(ShoeColor.objects.filter(
+                shoe__id = product.values('shoe__id').first()['shoe__id']
+            ).exclude(
+                id = product.values().first()['id']
+            ).annotate(
+                main_image = F('image__image')
+            ).values('id', 'main_image'))
+
+            color_list.insert(0, {
+                'id'         : shoe_detail[0]['id'],
+                'main_image' : shoe_detail[0]['main_image']
+            })
+
+            size_list = [size['shoecolorsize__size__name'] for size in product.values('shoecolorsize__size__name')]
+            shoe_detail.append({
+                'sub_image'  : sub_image,
+                'color_list' : color_list,
+                'size_list'  : size_list
+            })
+            return JsonResponse({'product' : shoe_detail}, status=200)
+        except IndexError:
+            return JsonResponse({'message' : 'NON_EXISTING_PRODUCT'}, status=400)
+        except ValueError:
+            return JsonResponse({'message' : 'VALUE_ERROR'}, status=400)
+
+class FilterView(View):
+    def get(self, request):
+        page = int(request.GET.get('page', 0))
+        limit = int(request.GET.get('limit', 20))
+        colorfilter  = request.GET.getlist('color', list(ColorFilter.objects.all().values_list('name', flat=True)))
+        typefilter   = request.GET.getlist('type', list(TypeFilter.objects.all().values_list('name', flat=True)))
+        genderfilter = request.GET.getlist('gender', list(GenderSegmentation.objects.all().values_list('name', flat=True)))
+        sizefilter   = request.GET.getlist('size', list(Size.objects.all().values_list('name', flat=True)))
+        
+        shoes = ShoeColor.objects.filter(**{
+            'color__color_category__name__in'     : colorfilter,
+            'shoe__type_filter__name__in'         : typefilter,
+            'shoe__gender_segmentation__name__in' : genderfilter,
+            'shoecolorsize__size__name__in'       : sizefilter,
+            'subimage__is_hovered'                : True
+        }).distinct()
         
-        return JsonResponse({'products' : shoe_list}, status=200)
+        shoe_values = list(shoes.annotate(
+            name       = F('shoe__detail__name'),
+            price      = F('shoe__price'),
+            main_image = F('image__image'),
+            sub_image  = F('subimage__image')
+        ).values('id','shoe__id', 'name', 'price', 'main_image', 'sub_image')[page*limit:((page+1)*limit)-1])
+
+        filters = {
+            'gender_filters'    : list(shoes.values_list('shoe__gender_segmentation__name',flat=True)),
+            'color_filters'     : [color.name for color in ColorFilter.objects.all()],
+            'type_filters'      : list(shoes.values_list('shoe__type_filter__name',flat=True)),
+            'size_filters'      : list(shoes.values_list('shoecolorsize__size__name',flat=True))
+        }
```

Comment:
아래는 제가 테스트 해본 코드인데 각필터값을 first 부분만 여러개 가져오도록 수정해서 테스트 부탁 드립니다.
```suggestion
            'gender_filters' : shoes.first().shoe.gender_segmentation.name
            'color_filters' : shoes.first().shoe.color.first().color_category.name
            'type_filters' : shoes.first().shoe.type_filter.name
            'size_filters' : shoes.first().size.first().name
```